lib/schemas.js

```diff
@@ -135,7 +135,7 @@ Schemas.prototype.cleanId = function (schema) {
 }
 
 Schemas.prototype.getSchemaAnyway = function (schema) {
-  if (schema.oneOf || schema.allOf || schema.anyOf) return schema
+  if (schema.oneOf || schema.allOf || schema.anyOf || schema.$merge || schema.$patch) return schema
```

Comment:
Do you think it would be possible to ask Ajv about its registered keywords?

@delvedor this is the most important change.

Reply:
Currently it can be complicated, need a lot of refactoring. The function `resolveRefs` is called before any instance of Ajv is available